service/tracker/execution.go

```diff
@@ -38,16 +40,68 @@ type Execution struct {
 
 // NewExecution creates a new DPS execution follower, relying on the provided
 // stream of block records (block data updates).
-func NewExecution(log zerolog.Logger, stream RecordStreamer) *Execution {
+func NewExecution(log zerolog.Logger, db *badger.DB, stream RecordStreamer) (*Execution, error) {
+
+	// The root block does not have a record that we can pull from the cloud
+	// stream of execution data. We thus construct it by getting the root block
+	// data from the DB directly.
+	var height uint64
+	err := db.View(operation.RetrieveRootHeight(&height))
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve root height: %w", err)
+	}
+	var blockID flow.Identifier
+	err = db.View(operation.LookupBlockHeight(height, &blockID))
+	if err != nil {
+		return nil, fmt.Errorf("could not look up root block: %w", err)
+	}
+	var header flow.Header
+	err = db.View(operation.RetrieveHeader(blockID, &header))
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve root header: %w", err)
+	}
+	var sealID flow.Identifier
+	err = db.View(operation.LookupBlockSeal(blockID, &sealID))
+	if err != nil {
+		return nil, fmt.Errorf("could not look up root seal: %w", err)
+	}
+	var seal flow.Seal
+	err = db.View(operation.RetrieveSeal(sealID, &seal))
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve root seal: %w", err)
+	}
 
 	s := Execution{
-		log:     log,
+		log:     log.With().Str("component", "execution_tracker").Logger(),
 		stream:  stream,
 		queue:   deque.New(),
 		records: make(map[flow.Identifier]*uploader.BlockData),
 	}
 
-	return &s
+	payload := flow.Payload{
```

Comment:
I have a hunch this is intentional, but I feel a comment pointing to the fact that the payload is empty serves a similar purpose as explicitly setting the values.

Reply:
I prefer to be explicit in these cases.